Sources/TuistSigning/SigningCipher.swift

```diff
@@ -0,0 +1,134 @@
+            .map { try encryptData($0, masterKey: masterKey) }
+
+        try zip(cipheredKeys, signingKeyFiles).forEach {
+            try $0.write(to: $1.url)
+        }
+    }
+
+    public func decryptSigning(at path: AbsolutePath) throws {
+        let (signingKeyFiles, masterKey) = try signingData(at: path)
+        let decipheredKeys = try signingKeyFiles
+            .map(FileHandler.shared.readFile)
+            .map {
+                try decryptData($0, masterKey: masterKey)
+            }
+
+        try zip(decipheredKeys, signingKeyFiles).forEach {
+            try $0.write(to: $1.url)
+        }
+    }
+
+    // MARK: - Helpers
+
+    /// Encrypts `data`
+    /// - Parameters:
+    ///     - data: Data to encrypt
+    ///     - masterKey: Master key data
+    /// - Returns: Encrypted data
+    private func encryptData(_ data: Data, masterKey: Data) throws -> Data {
+        let iv = try generateIv()
+        let aesCipher = try AES(key: masterKey.bytes, blockMode: CTR(iv: iv.bytes), padding: .noPadding)
+        guard
+            let encryptedBase64String = try aesCipher.encrypt(data.bytes).toBase64(),
+            let data = (iv.base64EncodedString() + "-" + encryptedBase64String).data(using: .utf8)
+        else { throw SigningCipherError.failedToEncrypt }
+        return data
+    }
+
+    /// Decrypts `data`
+    /// - Parameters:
+    ///     - data: Data to decrypt
+    ///     - masterKey: Master key data
+    /// - Returns: Decrypted data
+    private func decryptData(_ data: Data, masterKey: Data) throws -> Data {
+        guard
+            let encodedString = String(data: data, encoding: .utf8),
+            let dividerIndex = encodedString.firstIndex(of: "-"),
+            let iv = Data(base64Encoded: String(encodedString.prefix(upTo: dividerIndex)))
+        else { throw SigningCipherError.failedToEncrypt }
+
+        let dataToDecrypt = Data(base64Encoded: String(encodedString.suffix(from: dividerIndex).dropFirst()))
+        let aesCipher = try AES(key: masterKey.bytes, blockMode: CTR(iv: iv.bytes), padding: .noPadding)
+        guard let decryptedData = try dataToDecrypt?.decrypt(cipher: aesCipher) else { throw SigningCipherError.failedToEncrypt }
+        return decryptedData
+    }
+
+    /// - Returns: Files we want encrypt/decrypt along with master key data
+    private func signingData(at path: AbsolutePath) throws -> (signingKeyFiles: [AbsolutePath], masterKey: Data) {
+        guard
+            let rootDirectory = rootDirectoryLocator.locate(from: path)
+        else { throw SigningCipherError.rootDirectoryNotFound(path) }
+        let signingDirectory = rootDirectory.appending(components: Constants.tuistDirectoryName, Constants.signingDirectoryName)
+        let masterKey = try self.masterKey(from: signingDirectory)
+        // Find all files in `signingDirectory` with the exception of "master.key"
+        let signingKeyFiles = FileHandler.shared.glob(signingDirectory, glob: "**/*")
```

Comment:
I'd start by only encrypting/decrypting files under `Signing`. We want to be conventional about the structure of the directory and how the certificates and profiles are named and structure. If we follow the following convention there's no need to support subdirectories recursively:

```
Signing/
  Debug.cert.enc
  Release.cert.enc
  MyApp.Debug.mobileprovision.enc
  MyWatchApp.Debug.mobileprovision.enc
  master.key
```
Actually... after giving it another thought, I think we `master.key` can be under Tuist. That way we can use the key in the future to encrypt/decrypt other things. Something along the lines of:

```
master.key
Signing/
  Debug.cert.enc
  Release.cert.enc
  MyApp.Debug.mobileprovision.enc
  MyWatchApp.Debug.mobileprovision.enc
```

Reply:
Extract `master.key` and stopped parsing the subdirectories, your points make sense to me 👍